src/engine/attributes/datetime.js

```diff
@@ -1,16 +1,24 @@
 export default function dateOrDateTimeApply(val, userVal, op, discardTime){
 	let v = val
-
 	let uv = userVal
 
-	if(discardTime){
-		v = tsWithZeroTime(val)
-		uv = tsWithZeroTime(userVal)
+	if (uv === undefined || uv === null) {
+		throw new Error("Invalid date or datetime. Date or datetime must be type of date or time value in milliseconds")
 	}
-	
-	if(uv == null) {
-		// TODO log warning that name matches but type is not right
-		return false
+
+	if (uv.constructor.name == "Date") {
```

Comment:
This is added to support both type of input from user => Date and time in ms ?

Reply:
yeah, do you know what type nodejs developers use the most? Date or time in ms?

Reply:
It must be Date..